protocols/v2/binary-sv2/binary-sv2/src/lib.rs

```diff
@@ -1,4 +1,45 @@
-// TODO unify errors from serde_sv2 and no-serde-sv2
+//! Mediates between two implementations of the `binary_sv2` protocol,
+//! enabling encoding and decoding through `serde` or custom traits.
+//!
+//! # Overview
+//!
+//! Depending on the feature flags enabled, this crate will re-export implementations of the
+//! `Deserialize` and `Serialize` traits either from the `serde` library or from a custom,
+//! `serde`-free implementation provided by `binary_codec_sv2` and `derive_codec_sv2`. This allows
+//! for flexible integration of SV2 protocol types and binary serialization for environments that
+//! may not support `serde`.
+//!
+//! ## Features
+//! - **with_serde**: Enables `serde`-based serialization and deserialization for SV2 types,
+//!   using `serde` and `serde_sv2`.
+//! - **core**: Enables the custom `binary_codec_sv2` and `derive_codec_sv2` implementations,
+//!   which provide `Deserialize` and `Serialize` traits without the need for `serde`.
+//! - **prop_test**: Adds support for property testing for protocol types.
+//! - **with_buffer_pool**: Enables support for buffer pooling to optimize memory usage during
+//!   serialization and deserialization.
+//!
+//! ## Key Functions
+//!
+//! - `clone_message`: A utility function designed to clone messages of any type that implements
+//!   `Serialize`. This function is currently a placeholder and will be implemented to allow deep
+//!   cloning of SV2 protocol messages.
```

Comment:
I think there's no point in writing this.

This function should simply be marked for removal on https://github.com/stratum-mining/stratum/issues/1131